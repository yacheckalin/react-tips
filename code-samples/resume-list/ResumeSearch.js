import React, { useContext } from "react";
import PropTypes from "prop-types";
import ResumeListContext from "./ResumeListContext";

const ResumeSearch = ({ onSearch }) => {
  const { search } = useContext(ResumeListContext);
  return (
    <div className="col s4 input-field">
      <span className="prefix">
        <i className="material-icons green-text">search</i>
      </span>
      <input
        type="text"
        className="validate"
        value={search}
        onChange={(e) => {
          onSearch(e.target.value);
        }}
      />
    </div>
  );
};

ResumeSearch.propTypes = {
  onSearch: PropTypes.func.isRequired,
};

export default ResumeSearch;
