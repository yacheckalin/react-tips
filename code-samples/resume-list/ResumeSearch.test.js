import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

import ResumeSearch from "./ResumeSearch";

describe("<ResumeSearch />", () => {
  const props = {
    onSearch: jest.fn(),
  };
  test("renders and displays properly", () => {
    const wrapper = shallow(<ResumeSearch {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
