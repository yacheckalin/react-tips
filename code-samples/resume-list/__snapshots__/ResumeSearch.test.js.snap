// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResumeSearch /> renders and displays properly 1`] = `
<div
  className="col s4 input-field"
>
  <span
    className="prefix"
  >
    <i
      className="material-icons green-text"
    >
      search
    </i>
  </span>
  <input
    className="validate"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
